main.js

```diff
@@ -29,177 +29,173 @@
         _renderManager,
+        _assetManagers,
         _SONToCSSConverter;
 
-    var _assetManagers = {};
-
-    var _waitingDocuments = {},
-        _canceledDocuments = {};
-
     /**
-     * Disable asset generation for the given Document ID, halting any asset
-     * rending in progress.
-     * 
+     * Update the menu/state if the ID matches the current document.
+     * Changes to the asset generation status of a document may occur when it is not the active document in photoshop
+     *
      * @private
-     * @param {!number} id The document ID for which asset generation should be disabled.
+     * @param {number} id
+     * @param {boolean} checked
      */
-    function _pauseAssetGeneration(id) {
-        if (_waitingDocuments.hasOwnProperty(id)) {
-            _canceledDocuments[id] = true;
-        } else if (_assetManagers.hasOwnProperty(id)) {
-            _assetManagers[id].stop();
+    function _updateMenuIfActiveDoc(id, checked) {
+        if (id === _documentManager.getActiveDocumentID()) {
+            _stateManager.setState(id, checked);
         }
     }
 
     /**
-     * Completely stop asset generation for the given Document ID and collect
-     * its AssetManager instance.
-     * 
+     * Test if there an active AssetManager for the given document
+     *
      * @private
-     * @param {!number} id The document ID for which asset generation should be disabled.
+     * @param {number} id
+     * @return {boolean}
      */
-    function _stopAssetGeneration(id) {
-        _pauseAssetGeneration(id);
-
-        if (_assetManagers.hasOwnProperty(id)) {
-            delete _assetManagers[id];
-        }
+    function _documentIsGenerating(id) {
+        return _assetManagers.has(id);
     }
 
     /**
-     * Handler for a the "file" change event fired by Document objects. Disables
-     * asset generation after Save As is performed on an an already-saved file.
-     * 
+     * Start an AssetManager and add it to the local map of managers.
+     *
      * @private
-     * @param {number} id The ID of the Document that changed
-     * @param {{previous: string=, previousSaved: boolean=}}} change The file
-     *      change event emitted by the Document
+     * @param {Document} document
      */
-    function _handleFileChange(id, change) {
-        // If the filename changed but the saved state didn't change, then the file must have been renamed
-        if (change.previous && !change.hasOwnProperty("previousSaved")) {
-            _stopAssetGeneration(id);
-            _stateManager.deactivate(id);
+    function _startAssetManager(document) {
+        try {
+            var id = document.id,
+                assetManager = new AssetManager(_generator, _config, _logger, document, _renderManager);
+
+            _assetManagers.set(id, assetManager);
+
+            assetManager.once("idle", function () {
+                _logger.info("Asset generation complete", id);
+                // TODO need to destroy AM first?
```

Comment:
it seems like you did this TODO, can you update the comment to reflect that ( or tell me i'm reading the code wrong)